model/src/gateway/event/gateway.rs

```diff
@@ -307,13 +309,17 @@ impl<'de> Visitor<'de> for GatewayEventVisitor<'_> {
             "RECONNECT",
         ];
 
+        #[cfg(feature = "tracing")]
         let span = tracing::trace_span!("deserializing gateway event");
+        #[cfg(feature = "tracing")]
         let _span_enter = span.enter();
+        #[cfg(feature = "tracing")]
         tracing::trace!(event_type=?self.2, op=self.0, seq=?self.1);
```

Comment:
can these (and things like this) be in a block, like this? 
```suggestion
        #[cfg(feature = "tracing")]
        {
            let span = tracing::trace_span!("deserializing gateway event");
            let _span_enter = span.enter();
            tracing::trace!(event_type=?self.2, op=self.0, seq=?self.1);
        }
```

Reply:
They can't, because the span would be dropped too early and later tracing macro calls would no longer be inside the span.